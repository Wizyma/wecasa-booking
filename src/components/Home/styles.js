import styled from 'styled-components';

export const Grid = styled.div({
  width: '100%',
  display: 'flex',
  flexWrap: 'wrap',
  position: 'relative',
  justifyContent: 'center',
  textAlign: 'center',
});

export const GridColumn = styled.div({
  width: '300px',
  display: 'flex',
  flexWrap: 'wrap',
  boxShadow: '0 3px 6px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.23)',
  margin: '10px',
  padding: '10px',
  height: '150px',
  justifyContent: 'center',
  alignItems: 'center',
});

export const Paragraph = styled.p({
  margin: '2px',
  width: '100%',
});
