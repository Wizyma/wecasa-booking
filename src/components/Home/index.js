import React, { useContext, memo } from 'react';

import { Grid, GridColumn, Paragraph } from './styles';
import { formatPrice } from '../../helpers';
import Store from '../../context';
import Title from '../Title';

const Home = () => {
  const [store, call] = useContext(Store);
  const { ressources } = store;

  if (!ressources) return null;

  return (
    <Grid>
      <Title>{ressources.title}</Title>
      {ressources.categories.map(({ reference, title, prestations }) => (
        <Grid key={reference}>
          <Title variant="h2">{title}</Title>
          {prestations.map(({ duration, price, reference: ref, title: tit }) => (
            <GridColumn key={ref}>
              <Title variant="h4">{tit}</Title>
              <Paragraph>
                PRIX: <strong>{`${formatPrice(price)}â‚¬`}</strong>
              </Paragraph>
              <Paragraph>
                DUREE: <strong>{duration}</strong>
              </Paragraph>
              <button
                type="button"
                onClick={() =>
                  call({
                    type: 'ADD_TO_BASKEST',
                    payload: {
                      ref,
                      price: formatPrice(price),
                      duration,
                    },
                  })
                }
              >
                Ajouter prestations
              </button>
            </GridColumn>
          ))}
        </Grid>
      ))}
    </Grid>
  );
};

export default memo(Home);
