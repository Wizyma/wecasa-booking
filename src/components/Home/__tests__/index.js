import { render, cleanup } from 'react-testing-library';
import React from 'react';

import Home from '../';
import Store from '../../../context';

afterEach(cleanup);

describe('Homepage', () => {
  it('should render Homepage with mocked data', () => {
    const mockedData = {
      ressources: {
        categories: [
          {
            reference: 'man',
            title: 'Homme',
            prestations: [
              {
                reference: 'man_shampoo',
                title: 'Shampoing',
                duration: 10,
                price: 500,
              },
            ],
          },
        ],
        reference: 'haircut',
        title: 'Coiffure',
      },
      basket: null,
      appointment: null,
    };

    const tree = (
      <Store.Provider value={[mockedData]}>
        <Home />
      </Store.Provider>
    );
    const { baseElement } = render(tree);

    expect(baseElement).toMatchSnapshot();
  });

  it('should return null when no Data', () => {
    const tree = (
      <Store.Provider value={[{}]}>
        <Home />
      </Store.Provider>
    );
    const { baseElement } = render(tree);

    expect(baseElement).toMatchSnapshot();
  });
});
