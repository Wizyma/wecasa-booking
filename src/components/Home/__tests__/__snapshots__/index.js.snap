// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage should render Homepage with mocked data 1`] = `
<body>
  <div>
    <div
      class="sc-bdVaJa kBIsmA"
    >
      <h1
        class="sc-bxivhb gOekKA"
      >
        Coiffure
      </h1>
      <div
        class="sc-bdVaJa kBIsmA"
      >
        <h2
          class="sc-ifAKCX huOdvu"
        >
          Homme
        </h2>
        <div
          class="sc-bwzfXH fSBevD"
        >
          <h4
            class="sc-EHOje cKAoic"
          >
            Shampoing
          </h4>
          <p
            class="sc-htpNat kMGVrk"
          >
            PRIX: 
            <strong>
              50,0â‚¬
            </strong>
          </p>
          <p
            class="sc-htpNat kMGVrk"
          >
            DUREE: 
            <strong>
              10
            </strong>
          </p>
          <button
            type="button"
          >
            Ajouter prestations
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Homepage should return an empty div when no Data 1`] = `
<body>
  <div />
</body>
`;
